packages/wrangler/src/publish.ts

```diff
@@ -99,17 +98,41 @@ export default async function publish(props: Props): Promise<void> {
         jsxFragment,
         rules: props.rules,
         tsconfig: props.tsconfig ?? config.tsconfig,
+        durable_objects: config.durable_objects,
       }
     );
 
+    // Some validation of durable objects + migrations
+    if (config.durable_objects.bindings.length > 0) {
+      // TODO: implement durable objects for service environments
+      if (!props.legacyEnv) {
+        throw new Error(
+          "Publishing Durable Objects to a service environment is not currently supported. We'll fix this soon!"
+        );
+      }
+
+      // intrinsic [durable_objects] implies [migrations]
+      if (
+        config.durable_objects.bindings.filter(
+          (binding) => !binding.script_name
+        ).length > 0 &&
+        config.migrations.length === 0
+      ) {
+        console.warn(
```

Comment:
This is a warning because folks may have deleted the `[migrations]` section from their `wrangler.toml` altogether.